synapse/util/ratelimitutils.py

```diff
@@ -154,7 +154,7 @@ def ratelimit(self, host: str) -> "Iterator[defer.Deferred[None]]":
         self.host = host
 
         request_id = object()
-        ret = self._on_enter(request_id)
+        ret = defer.ensureDeferred(self._on_enter_with_tracing(request_id))
```

Comment:
We can add this comment copied from: https://github.com/matrix-org/synapse/blob/49d04e43dfa0551f52c1f1872b6f311efa756ca8/synapse/util/async_helpers.py#L760-L761

Because the Twisted docs mention:

> The [`ensureDeferred`](https://docs.twistedmatrix.com/en/stable/api/twisted.internet.defer.html#ensureDeferred) function also provides a way to convert a coroutine to a Deferred, but it’s interface is more type-ambiguous; `Deferred.fromCoroutine` is meant to replace it.
>
> *-- https://docs.twistedmatrix.com/en/stable/core/howto/defer-intro.html#coroutines-with-async-await*